Sources/LeafKit/LeafToken.swift

```diff
@@ -196,3 +213,100 @@ public enum LeafToken: CustomStringConvertible, Equatable  {
         }
     }
 }
+
+internal extension Array where Element == ParameterDeclaration {
+    // evaluate a flat array of Parameters ("Expression")
+    // returns true if the expression was reduced, false if
+    // not or if unreducable (eg, non-flat or no operands).
+    // Does not promise that the resulting Expression is valid.
+    // This is brute force and not very efficient.
```

Comment:
### Architecture Notes ###
This is the logic driving operand grouping. It can definitely be enhanced, but a model like this may be better suited than ParameterResolver to efficiency as building a tuple map of operands allows straight like-to-like comparisons on equatable lhs/rhs input before burying them inside Syntax and having to verify everything at the serialize stage.

This also would allow for pre-serialization static variables to be injected and compressed in the AST before caching, and for Parser to expose a Set of named Variables and Tags that can be attached to the AST so the application can query what's expected/asked for before attempting to render.